modules/home-manager/lsd.nix

```
@@ -0,0 +1,24 @@
+{ catppuccinLib }:
+{ config, lib, ... }:
+
+let
+  inherit (config.catppuccin) sources;
+  cfg = config.catppuccin.lsd;
+in
+
+{
+  options.catppuccin.lsd = catppuccinLib.mkCatppuccinOption { name = "lsd"; };
+
+  imports = catppuccinLib.mkRenamedCatppuccinOptions {
+    from = [
+      "programs"
+      "lsd"
+      "catppuccin"
+    ];
+    to = "lsd";
+  };
+
+  config = lib.mkIf cfg.enable {
+    programs.lsd.colors = catppuccinLib.importYAML "${sources.lsd}/catppuccin-${cfg.flavor}/colors.yaml";
```

```suggestion
    xdg.configFile = {
      "lsd/colors.yaml".source = "${sources.lsd}/catppuccin-${cfg.flavor}/colors.yaml";
    }
```

I think you can do this to avoid IFD. https://github.com/lsd-rs/lsd#color-theme

---

That was my initial thought as well. However, upon discovering the [`programs.lsd.colors`](https://nix-community.github.io/home-manager/options.xhtml#opt-programs.lsd.colors) option, I decided it would be more convenient to leverage it for the implementation.

In any case, with your suggested approach, do you think we should explicitly set
```nix
programs.lsd.settings.color.theme = "custom";
```
to ensure LSD uses the colors.yaml file? Note that this step is handled automatically when using the `lsd.colors` option.

---

I would prefer to remove IFD with the `colours.yml` file. we should also apply the theme as you suggested.

---

I've committed the changes. Let me know if it is to your liking.